src/setup.js

```diff
@@ -159,13 +191,13 @@ function parseSingletons(singletons){
     return singletonMapping;
 }
 
-function generator(dependence, entities, complexTypes, enums, entitySets, singletons){
+function generator(dependence, entities, complexTypes, enums, entitySets, singletons, entityAnnotations, complexAnnotations){
```

Comment:
Maybe just have a single annotations parameter that's shared for all types? It's currently entity and complex but might be more later and then the number of params will keep increasing. I think that's a safe operation since `generateTypeDefs` checks if it's in the dictionary and ignores anything else

Reply:
I like this idea. I agree that this should be safe assuming there are no conflicting names between enums, entity and complex types.